tests/fullstack-test2/variables/set_sql_mode_ansi.test

```diff
@@ -0,0 +1,33 @@
+# Copyright 2024 PingCAP, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# use an empty string as the default value for decimal column
+mysql> set sql_mode='ansi'; create table test.t (a int, d decimal(34, 2) default '', d2 decimal(34, 2) default 'abcd');
+
+mysql> alter table test.t set tiflash replica 1;
+
+func> wait_table test t;
+
+mysql> set sql_mode='ansi'; insert into test.t (a) values (1),(2),(3),(4);
```

Comment:
I think we should add these rows before adding tiflash replica. Or the default value parsing may be skipped on apply snapshot stage?